src/navigations/HomeBottomTabs.js

```diff
@@ -207,135 +211,116 @@ function HomeBottomTabs({navigation}) {
+            route.name === 'AnonymousChannelList' ? colors.anon_primary : colors.darkBlue
         })}>
         <Tab.Screen
-          name="Feed"
-          component={FeedScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        />
-        {/* <Tab.Screen
-          name="Feed"
-          component={WebsocketResearchScreen}
-          initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Feed')
-            // unmountOnBlur: true
-          }}
-        /> */}
-        {/* <Tab.Screen
-          name="ChannelList"
+          name="SignedChannelList"
           component={ChannelListScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge:
-              unReadMessage.total_unread_count + unReadMessage.unread_post > 0
-                ? unReadMessage.total_unread_count + unReadMessage.unread_post
-                : null
-          }}
-        /> */}
+          listeners={({route}) => saveLastMenu(route)}
+          options={{tabBarBadge: signedChannelUnreadCount > 0 ? signedChannelUnreadCount : null}}
+        />
         <Tab.Screen
-          name="ChannelList"
+          name="AnonymousChannelList"
           component={ChannelListScreenV2}
           initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
           options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('ChannelList'),
-            tabBarBadge: totalUnreadCount > 0 ? totalUnreadCount : null
+            tabBarBadge: anonymousChannelUnreadCount > 0 ? anonymousChannelUnreadCount : null
           }}
         />
+        <Tab.Screen
+          name="Feed"
+          component={FeedScreen}
+          initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
+        />
         <Tab.Screen
           name="News"
           component={NewsScreen}
           initialParams={{isBottomTab: true}}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('News')
-            // unmountOnBlur: true
-          }}
+          listeners={({route}) => saveLastMenu(route)}
         />
         <Tab.Screen
           name="Profile"
-          initialParams={{isBottomTab: true}}
           component={ProfileScreen}
-          options={{
-            activeTintColor: colors.holytosca,
-            tabBarIcon: renderTabLabelIcon('Profile')
-            // unmountOnBlur:true
-          }}
+          initialParams={{isBottomTab: true}}
+          listeners={({route}) => saveLastMenu(route)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for setting the initial route name and saving the last selected menu has been refactored. The changes seem to improve the code's readability and maintainability. However, ensure that the `saveLastMenu` function is working as expected and updating the storage with the correct route name.
```diff
-  const lastMenu = StorageUtils.lastSelectedMenu.get();
-  const getInitialRouteName = React.useCallback(() => {
-    if (initialStartup !== null && otherProfileData?.user_id === initialStartup.id) {
-      return 'Profile';
-    }
-
-    return lastMenu ?? 'Feed';
-  }, [lastMenu]);
-
-  const saveLastMenu = (route) => ({
-    tabPress: () => {
-      StorageUtils.lastSelectedMenu.set(route?.name);
-    }
-  });
+  const lastMenu = StorageUtils.lastSelectedMenu.get();
+  const getInitialRouteName = React.useCallback(() => {
+    if (initialStartup !== null && otherProfileData?.user_id === initialStartup.id) {
+      return 'Profile';
+    }
+
+    return lastMenu ?? 'Feed';
+  }, [lastMenu]);
+
+  const saveLastMenu = (route) => ({
+    tabPress: () => {
+      StorageUtils.lastSelectedMenu.set(route?.name);
+    }
+  });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->